### PR TITLE
Packmol

### PR DESCRIPTION
**Pull request by chrisbarnettster**
Create simulation boxes using packmol.

### CODE REVIEW

**Comment by chrisbarnettster** [comment]
@bgruening test is failing because packmol is currently in the omnia channel. Can I specify a custom channel, or should I try get it to be in a best practice channel (or a channel I manage)?

**Comment by bgruening** [comment]
@chrisbarnettster, in theory, you can change the channel order. In practice, I would try to get it into conda-forge.

File: tools/buildtools/packmol/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# Packmol for Galaxy
+
+# See the packmol user guide for more
+http://m3g.iqm.unicamp.br/packmol/userguide.shtml#more
+
+# calculating the number of molecules ise the volume guesser @
+http://m3g.iqm.unicamp.br/packmol/utilities.shtml
+
+# run test
+planemo test --conda_dependency_resolution --conda_prefix /scratch/cbarnett/conda_for_planemo/ galaxy_packmol.xml
```
Comment (bgruening) [inline tools/buildtools/packmol/README.md]: --conda_dependency_resuolution should be the default by now.

I would remove this file I think

File: tools/buildtools/packmol/packmol.xml [hunk 2]
```diff
@@ -0,0 +1,196 @@
+<tool id="packmol" name="PACKMOL" version="1.18.013">
+    <description>Initial configurations for Molecular Dynamics Simulations by packing optimization</description>
+    <macros>
+        <import>macros_packmol.xml</import>
+    </macros>
+    <requirements>
+      <requirement type="package" version="1!18.013">packmol</requirement>
```
Comment (bgruening) [inline tools/buildtools/packmol/packmol.xml]: wee, what is this, do they really have a ! in the version?

File: tools/buildtools/packmol/packmol.xml [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    </requirements>
+    <command detect_errors="exit_code">
+        <![CDATA[
+        python "$packmol_script" '$inputs'
```
Comment (bgruening) [inline tools/buildtools/packmol/packmol.xml]: please single quotes

File: tools/buildtools/packmol/packmol.xml [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+
+currentpath = "$__tool_directory__"  # should work generally
+template_environment = Environment(loader=FileSystemLoader(currentpath),lstrip_blocks=True, trim_blocks=True)
+template = template_environment.get_template('template.j2')
+print(params)
+
+with open("$parameteroutfile",'w+') as f:
+    f.write(template.render(params))
+
+os.system("packmol < "+"$parameteroutfile")
```
Comment (bgruening) [inline tools/buildtools/packmol/packmol.xml]: os.system is no really recommend to used theses days. I think you can remove this call here and put this straight as `packmol < '$parameteroutfile'` into the cheetah-command above.

File: tools/buildtools/packmol/packmol.xml [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+                <expand macro="multiple_input" name="packmol_inputs" format="xyz" label="xyz input with molecule type" help_text="Select a xyz file in .xyz format"/>
+            </when>
+            <when value="tinker">
+                <expand macro="multiple_input" name="packmol_inputs" format="tinker" label="tinker input with molecule type" help_text="Select a tinker file in .tinker format"/>
+            </when>
+            <when value="moldy">
+                <expand macro="multiple_input" name="packmol_inputs" format="moldy" label="moldy input with molecule type" help_text="Select a moldy file in .moldy format"/>
+            </when>
+        </conditional>
+    <section name="allparams" title="General Parameters" expanded="false">
```
Comment (bgruening) [inline tools/buildtools/packmol/packmol.xml]: indentation

File: tools/buildtools/packmol/packmol.xml [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+        <param name="tolerance" type="float" value="2.0" label="distance tolerance" />
+        <param name="nloop" type="integer" value="20" label="maximum number of optimisation loops" />
```
Comment (bgruening) [inline tools/buildtools/packmol/packmol.xml]: do you have any idea about min/max values of all these parameters?

File: tools/buildtools/packmol/packmol.xml [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+
+
+    </inputs>
+    <outputs>
+        <data format="pdb" name="outfile" label="${tool.name}: structure output">
+        <change_format>
+            <when input="selected_format" value="pdb" format="pdb"/>
+            <when input="selected_format" value="xyz" format="xyz"/>
+            <when input="selected_format" value="tinker" format="tinker"/>
+            <when input="selected_format" value="moldy" format="moldy"/>
```
Comment (bgruening) [inline tools/buildtools/packmol/packmol.xml]: are these filytypes alls in Galaxy?

File: tools/buildtools/packmol/packmol.xml [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+        </data>
+        <data format="txt" name="parameteroutfile" label="${tool.name}: parameter output"/>
+    </outputs>
+    <tests>
+        <test>
+            <param name="selected_format" value="pdb"/>
+            <param name="packmol_inputs_0|input" value="water.pdb" ftype="pdb"/>
+            <param name="allparams.seed" value="101"/>
+            <output name="outfile" file="packmol_outfile1.pdb" compare="sim_size" delta="100000">
+             <assert_contents>
```
Comment (bgruening) [inline tools/buildtools/packmol/packmol.xml]: indentation

File: tools/buildtools/packmol/packmol.xml (hunk 2 above)
Comment (chrisbarnettster) [reply tools/buildtools/packmol/packmol.xml]: yup that is someone else's package!

File: tools/buildtools/packmol/packmol.xml (hunk 7 above)
Comment (chrisbarnettster) [reply tools/buildtools/packmol/packmol.xml]: only pdb at the moment - so those extras don't work yet. I can comment out?

**Comment by bgruening** [comment]
@chrisbarnettster any help needed for the conda package?

File: tools/buildtools/packmol/packmol.xml [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+.. class:: infomark
+
+**User guide and documentation**
+
+- Packmol `userguide`_
+- Calculating the number of molecules using the `volume guesser`_
+
+.. class:: infomark
+
+**Features not included**
```
Comment (bgruening) [inline tools/buildtools/packmol/packmol.xml]: Should we move this into a github issue or a readme? I'm not sure we should expose this to a user.

Comment (chrisbarnettster) [reply tools/buildtools/packmol/packmol.xml]: https://github.com/galaxycomputationalchemistry/galaxy-tools-compchem/issues/8

Comment (chrisbarnettster) [reply tools/buildtools/packmol/packmol.xml]: agreed, added a github issue.

File: tools/buildtools/packmol/packmol.xml (hunk 7 above)
Comment (chrisbarnettster) [reply tools/buildtools/packmol/packmol.xml]: commenting out and committing in a sec

**Pull request merged**